src/lib.rs

```diff
@@ -1682,14 +1682,48 @@ fn range_to_offset_size<S: RangeBounds<BufferAddress>>(
         Bound::Unbounded => 0,
     };
     let size = match bounds.end_bound() {
-        Bound::Included(&bound) => BufferSize::new(bound + 1 - offset),
-        Bound::Excluded(&bound) => BufferSize::new(bound - offset),
+        Bound::Included(&bound) => Some(bound + 1 - offset),
+        Bound::Excluded(&bound) => Some(bound - offset),
         Bound::Unbounded => None,
-    };
+    }
+    .map(|size| {
```

Comment:
why not just `BufferSize::new(size).expect("Buffer slices can not be empty")`?

Reply:
Because I still don't have all the Option utility functions in my head 😊 
Sooo much better :)